From 1ad3d3b99c680483d1ee1eec2b01010492bee9bb Mon Sep 17 00:00:00 2001
From: Shengzhou <shengzhou.liu@nxp.com>
Date: Thu, 10 Oct 2024 10:19:58 +0200
Subject: [PATCH] fix build issues to support compiling gst_plugins_bad

support compiling gst_plugins_bad 1.24.x with meson lower version 0.61.2 on Ubuntu 22.04 host rather than the required meson v1.1+
---
 ext/dtls/meson.build         | 29 -----------------
 ext/meson.build              |  3 --
 ext/sctp/meson.build         | 62 ------------------------------------
 ext/srtp/meson.build         | 45 --------------------------
 gst-libs/gst/mse/meson.build |  1 -
 meson.build                  |  4 ++-
 tests/check/meson.build      |  2 --
 tests/meson.build            |  2 --
 8 files changed, 3 insertions(+), 145 deletions(-)
 delete mode 100644 ext/dtls/meson.build
 delete mode 100644 ext/sctp/meson.build
 delete mode 100644 ext/srtp/meson.build

diff --git a/ext/dtls/meson.build b/ext/dtls/meson.build
deleted file mode 100644
index ca07da01e..000000000
--- a/ext/dtls/meson.build
+++ /dev/null
@@ -1,29 +0,0 @@
-dtls_sources = [
-  'gstdtlsagent.c',
-  'gstdtlscertificate.c',
-  'gstdtlsconnection.c',
-  'gstdtlsdec.c',
-  'gstdtlsenc.c',
-  'gstdtlssrtpbin.c',
-  'gstdtlssrtpdec.c',
-  'gstdtlssrtpdemux.c',
-  'gstdtlssrtpenc.c',
-  'plugin.c',
-  'gstdtlselement.c',
-]
-
-dtls_option = get_option('dtls').enable_if(get_option('webrtc').enabled(), error_message: 'webrtc option is enabled')
-openssl_dep = dependency('openssl', version: '>= 1.0.1', required: dtls_option)
-libcrypto_dep = dependency('libcrypto', required: dtls_option)
-
-if openssl_dep.found() and libcrypto_dep.found()
-  gstdtls = library('gstdtls',
-    dtls_sources,
-    c_args : gst_plugins_bad_args,
-    include_directories : [configinc],
-    dependencies : [gst_dep, libcrypto_dep, openssl_dep] + winsock2,
-    install : true,
-    install_dir : plugins_install_dir,
-  )
-  plugins += [gstdtls]
-endif
diff --git a/ext/meson.build b/ext/meson.build
index 4f7139835..d3bc04814 100644
--- a/ext/meson.build
+++ b/ext/meson.build
@@ -13,7 +13,6 @@ subdir('curl')
 subdir('dash')
 subdir('dc1394')
 subdir('directfb')
-subdir('dtls')
 subdir('dts')
 subdir('faac')
 subdir('faad')
@@ -54,13 +53,11 @@ subdir('resindvd')
 subdir('rsvg')
 subdir('rtmp')
 subdir('sbc')
-subdir('sctp')
 subdir('smoothstreaming')
 subdir('sndfile')
 subdir('soundtouch')
 subdir('spandsp')
 subdir('srt')
-subdir('srtp')
 subdir('svtav1')
 subdir('svthevcenc')
 subdir('teletextdec')
diff --git a/ext/sctp/meson.build b/ext/sctp/meson.build
deleted file mode 100644
index 2b4f88a86..000000000
--- a/ext/sctp/meson.build
+++ /dev/null
@@ -1,62 +0,0 @@
-sctp_sources = [
-  'gstsctpdec.c',
-  'gstsctpenc.c',
-  'gstsctpplugin.c',
-  'sctpassociation.c'
-]
-
-sctp_option = get_option('sctp').enable_if(get_option('webrtc').enabled(), error_message: 'webrtc option is enabled')
-if sctp_option.disabled()
-  subdir_done()
-endif
-
-sctp_platform_deps = []
-
-found_system_usrsctp = false
-
-if not get_option('sctp-internal-usrsctp').enabled()
-  sctp_dep = cc.find_library('usrsctp', required: false)
-  sctp_header = cc.has_header('usrsctp.h')
-  if host_system == 'windows'
-    sctp_platform_deps += [cc.find_library('ws2_32')]
-  endif
-
-  found_system_usrsctp = sctp_dep.found() and sctp_header
-
-  if get_option('sctp-internal-usrsctp').disabled() and not found_system_usrsctp
-    if sctp_option.enabled()
-      error('sctp plugin enabled but could not find libusrsctp or usrsctp.h, and internal libusrsctp disabled')
-    else
-      message('Could not find libusrsctp or usrsctp.h, and internal libusrsctp disabled - not building sctp plugin')
-      subdir_done()
-    endif
-  endif
-endif
-
-if not found_system_usrsctp
-  message('Using internal libusrsctp')
-  subdir('usrsctp')
-  sctp_dep = usrsctp_dep
-  sctp_header = true
-  if sctp_option.enabled() and not sctp_dep.found()
-    error('sctp plugin enabled but could not find system libusrsctp or configure internal libusrsctp')
-  endif
-endif
-
-if not gst_debug_disabled
-  sctp_args = ['-DSCTP_DEBUG']
-else
-  sctp_args = []
-endif
-
-if sctp_dep.found() and sctp_header
-  gstsctp = library('gstsctp',
-    sctp_sources,
-    c_args : gst_plugins_bad_args + sctp_args,
-    include_directories : [configinc],
-    dependencies : [sctp_dep, gst_dep, gstbase_dep, gstsctp_dep, sctp_platform_deps],
-    install : true,
-    install_dir : plugins_install_dir,
-  )
-  plugins += [gstsctp]
-endif
diff --git a/ext/srtp/meson.build b/ext/srtp/meson.build
deleted file mode 100644
index d15d58c13..000000000
--- a/ext/srtp/meson.build
+++ /dev/null
@@ -1,45 +0,0 @@
-srtp_sources = [
-  'gstsrtp.c',
-  'gstsrtpelement.c',
-  'gstsrtpplugin.c',
-  'gstsrtpdec.c',
-  'gstsrtpenc.c',
-]
-
-srtp_cargs = []
-srtp_option = get_option('srtp').enable_if(get_option('webrtc').enabled(), error_message: 'webrtc option is enabled')
-if srtp_option.disabled()
-  srtp_dep = dependency('', required : false)
-  subdir_done()
-endif
-
-srtp_dep = dependency('libsrtp2', version : '>= 2.1.0', required : false, allow_fallback: true)
-if srtp_dep.found()
-  srtp_cargs += ['-DHAVE_SRTP2']
-else
-  srtp_dep = dependency('libsrtp', version: '>= 1.6.0', required : false)
-  if not srtp_dep.found() and cc.has_header_symbol('srtp/srtp.h', 'crypto_policy_set_aes_gcm_128_16_auth')
-    srtp_dep = cc.find_library('srtp', required : false)
-  endif
-endif
-if not srtp_dep.found() and srtp_option.enabled()
-  error('srtp plugin enabled but libsrtp not found')
-endif
-
-if srtp_dep.found()
-  gstsrtp_enums = gnome.mkenums_simple('gstsrtp-enumtypes',
-    sources : ['gstsrtpenums.h'],
-    decorator : 'G_GNUC_INTERNAL',
-    install_header: false)
-
-  gstsrtp = library('gstsrtp',
-    srtp_sources, gstsrtp_enums,
-    c_args : gst_plugins_bad_args + srtp_cargs,
-    link_args : noseh_link_args,
-    include_directories : [configinc],
-    dependencies : [gstrtp_dep, gstvideo_dep, srtp_dep],
-    install : true,
-    install_dir : plugins_install_dir,
-  )
-  plugins += [gstsrtp]
-endif
diff --git a/gst-libs/gst/mse/meson.build b/gst-libs/gst/mse/meson.build
index ebc5c337b..952503439 100644
--- a/gst-libs/gst/mse/meson.build
+++ b/gst-libs/gst/mse/meson.build
@@ -84,7 +84,6 @@ gstmse_private_test = library('gstmse-private-test',
 )
 
 gstmse_private_test_dep = declare_dependency(
-  compile_args : [gstmse_c_args, '-DBUILDING_GST_MSE_TEST'],
   link_with : [gstmse_private_test],
   include_directories : [libsinc],
   sources: gstmse_headers_public + [gstmse_enums_h],
diff --git a/meson.build b/meson.build
index a0eb1dbc4..032799a9d 100644
--- a/meson.build
+++ b/meson.build
@@ -1,6 +1,6 @@
 project('gst-plugins-bad', 'c', 'cpp',
   version : '1.24.0',
-  meson_version : '>= 1.1',
+  meson_version : '>= 0.61',
   default_options : [ 'warning_level=1',
                       'buildtype=debugoptimized' ])
 
@@ -537,6 +537,8 @@ if get_option('default_library') == 'shared'
 endif
 
 pkgconfig_variables = ['exec_prefix=${prefix}',
+  'datadir={prefix}/share',
+  'libexecdir={prefix}/libexec',
     'toolsdir=${exec_prefix}/bin',
     'pluginsdir=${libdir}/gstreamer-1.0',
     'datarootdir=${prefix}/share',
diff --git a/tests/check/meson.build b/tests/check/meson.build
index c21887d99..329df4dc7 100644
--- a/tests/check/meson.build
+++ b/tests/check/meson.build
@@ -70,7 +70,6 @@ base_tests = [
   [['elements/rtpsink.c'], get_option('rtp').disabled()],
   [['elements/sdpdemux.c'], get_option('sdp').disabled(), [gstsdp_dep]],
   [['elements/srt.c'], not srt_dep.found(), [srt_dep]],
-  [['elements/srtp.c'], not srtp_dep.found(), [srtp_dep]],
   [['elements/switchbin.c'], get_option('switchbin').disabled()],
   [['elements/videoframe-audiolevel.c'], get_option('videoframe_audiolevel').disabled()],
   [['elements/viewfinderbin.c']],
@@ -135,7 +134,6 @@ if host_machine.system() != 'windows'
     [['elements/curlftpsink.c'], not curl_dep.found(), [curl_dep]],
     [['elements/curlsmtpsink.c'], not curl_dep.found(), [curl_dep]],
     [['elements/dash_mpd.c'], not xml2_dep.found(), [xml2_dep]],
-    [['elements/dtls.c'], not libcrypto_dep.found() or not openssl_dep.found (), [libcrypto_dep]],
     [['elements/faac.c'],
         not faac_dep.found() or not cc.has_header_symbol('faac.h', 'faacEncOpen') or not cdata.has('HAVE_UNISTD_H'),
         [faac_dep]],
diff --git a/tests/meson.build b/tests/meson.build
index 1118a060d..51ae86aac 100644
--- a/tests/meson.build
+++ b/tests/meson.build
@@ -3,9 +3,7 @@ if get_option('tests').disabled() or static_build
 endif
 
 if gstcheck_dep.found()
-  subdir('check')
   subdir('interactive')
-  subdir('validate')
 endif
 if not get_option('examples').disabled()
   subdir('examples')
-- 
2.34.1

